src/main/java/io/vavr/collection/Iterator.java

```diff
@@ -742,39 +742,48 @@ public BigDecimal next() {
         } else if (Integer.signum(step) == Integer.signum(from - toInclusive)) {
```

Comment:
@mincong-h this will still fail for `from = Integer.MAX_VALUE`, `toInclusive = -5` and `step = -1`.

Reply:
It would be better to completely remove the `if (Integer.signum(step) == Integer.signum(from - toInclusive))` check.

Instead we could perform these checks in the `step > 0` and `step < 0` if branches:

```java
if (step == 0) {
    throw new IllegalArgumentException("step cannot be 0");
} else if (from == toInclusive) {
    return of(from);
} if (step > 0) {
    if (from > toInclusive) {
        return empty();
    } else {
        return new Iterator<Integer>() { ... };
    }
} else {
    if (from < toInclusive) {
        return empty();
    } else {
        return new Iterator<Integer>() { ... };
    }
}    
```

Reply:
Yes, thanks for your hint. I'm working on a fix now. Will add test(s) for this as well.